### PR TITLE
chore: TS declaration map files

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Enable declaaration map on typescript builds, so navigation in your code editor links to actual source files, not the dist built files.

### CODE REVIEW

File: packages/core/data-transfer/package.json [hunk 1]
```diff
@@ -26,19 +26,19 @@
   ],
   "main": "./dist/index.js",
   "types": "./dist/index.d.ts",
+  "files": [
+    "./dist/**/!(*.d.ts.map)"
```
Comment (joshuaellis) [inline packages/core/data-transfer/package.json:30]: no need to show off

**Review by joshuaellis — COMMENTED** [review]
would it not make more sense to include this in the `base` and `client` config instead?

File: packages/core/data-transfer/package.json (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/data-transfer/package.json:30]: for some reason my editor decided to reorder the package.json files

**Comment by Marc-Roig** [comment]
don't we only need those files during the build time? (which is the client config btw?)

**Comment by joshuaellis** [comment]
> don't we only need those files during the build time? (which is the client config btw?)

the `base` is what we extend all `tsconfig`s from it lives in `packages/utils/tsconfig` as does the client – and it doesn't really matter imo because the build configs always extend the original

**Comment by Marc-Roig** [comment]
ah! that makes much more sense . Will update the base one :pepe-run:

**Comment by Marc-Roig** [comment]
done! moved those configs to the base.json and client.json base files 🚀

**Comment by Convly** [comment]
Seems like the build is failing :eyes: 
Could you share the output of npm pack for one of the built package just for context

**Comment by Marc-Roig** [comment]
looks like `declarationMap` config can only be set in the `tsconfig` if also setting the `declaration` config. 

Not sure why it's not set in the `client.json` but it's set for the `base.json` file. Do you know why? @joshuaellis @alexandrebodin

**Comment by joshuaellis** [comment]
ahhh, it's not set in the `client` config because we use a `Bundler` because we're FE packages which means TS does not expect to be producing any types 🤓

maybe just put it in Base for now and we can reconcile later... with pack-up when all the packages have been moved we can just add it to the library either as a setting or a default idm

**Review by joshuaellis — DISMISSED** [review]

**Review by joshuaellis — APPROVED** [review]

**Review by innerdvations — APPROVED** [review]
LGTM thank you for this!

**Pull request merged**